### DISCUSSION TITLE
React

### DISCUSSION THREAD

**Discussion by chris-aeviator** [Point72/raydar#49] [Ideas]
Currently there are some interface pains with raydar and the interaction with perspective (e.g. dropdowns are finicky, `create table` only shows one item w/o scrolling, theme selection only works after multiple clicks). To ensure the stability of UI I'd like to raise the discussion if a react (or react-like) ui framework is intended to be part of raydar. I understand that raydar is at it's infancy and other features might have a higher priority.

I'd like to contribute a next.js implementation if nothing speaks against this from your side. In general and IMO vanilla JS based development is not very scalable.

**Comment by timkpaine** [comment]
> dropdowns are finicky, `create table` only shows one item w/o scrolling, theme selection only works after multiple clicks

I can't reproduce any of these, please submit a bug report if you have a minimal reproducible example of this behavior (a screen recording would be good too).

React and perspective are orthogonal concepts, there is currently no plan for any frontend framework but if we added one, it likely would not be react as we don't really need 99% of the benefit react would bring.

**Reply by chris-aeviator** [reply]
- https://github.com/Point72/raydar/issues/50
- https://github.com/Point72/raydar/issues/51

unsure why you'd ask for a minimal reproducible example, this is all general UI, I attached screencasts to both of them 

the third issue (multiple clicks for triggering a theme change) only happens on my linux machine that I can't record videos on RN.

**Comment by gauglertodd** [comment]
hey @chris-aeviator, thanks for the idea! From your perspective: what would be the biggest advantages of react over the current impl?
